### ISSUE TITLE
PyTorch normalizing flows 

### ISSUE THREAD

**Issue by ncfrey** [deepchem/deepchem#2763]
## 🚀 Feature

A PyTorch implementation of normalizing flows for deep generative modeling and distribution learning.

## Motivation

The current implementation for normalizing flows in DeepChem is written in TensorFlow Probability. Following the general tend of increasing PyTorch adoption in DeepChem, it would be useful to have the core classes of the normalizing flows API written in PyTorch.

## Additional context
The [nflows](https://github.com/bayesiains/nflows) library provides good base classes for distributions and bijectors for normalizing flows.

**Comment by rbharath** [comment]
Thanks for writing this up! This might be a good project for GSoC next summer if we are selected to participate

**Comment by rbharath** [comment]
I believe this has been completed by @JoseAntonioSiguenza . Are we good to close this issue?

**Comment by JoseAntonioSiguenza** [comment]
> I believe this has been completed by @JoseAntonioSiguenza . Are we good to close this issue?

Yes! There are some features to add, but this can be addressed in a new issue.  🙌🏼
